test/src/e2e_vm_tests/test_programs/should_pass/static_analysis/cei_pattern_violation/src/main.sw

```diff
@@ -0,0 +1,34 @@
+contract;
+
+use std::storage::store;
+
+// TODO when the test harness can expect warnings, _this should warn about CEI_!
+
+abi TestAbi {
+  #[storage(write)]
+  fn deposit(gas: u64, amount: u64, color: b256,  unused: ());
+}
+
+#[storage(write)]
+fn deposit(gas: u64,  amount: u64, color: b256, unused: ()) {
+let other_contract = abi(TestAbi, 0x3dba0a4455b598b7655a7fb430883d96c9527ef275b49739e7b0ad12f8280eae);
+
+// interaction
+other_contract.deposit(gas, amount, color, unused);
+// [toplevel function] effect -- therefore violation of CEI where effect should go before interaction
+let storage_key = 0x3dba0a4455b598b7655a7fb430883d96c9527ef275b49739e7b0ad12f8280eae;
+//store(storage_key, ());
+}
+
+impl TestAbi for Contract {
+  #[storage(write)]
+  fn deposit(gas: u64,  amount: u64, color: b256, unused: ()) {
+    let other_contract = abi(TestAbi, 0x3dba0a4455b598b7655a7fb430883d96c9527ef275b49739e7b0ad12f8280eae);
+
+    // interaction
+    other_contract.deposit(gas, amount, color, unused);
+    // [trait/abi method] effect -- therefore violation of CEI where effect should go before interaction
+    let storage_key = 0x3dba0a4455b598b7655a7fb430883d96c9527ef275b49739e7b0ad12f8280eae;
+    store(storage_key, ());
```

Comment:
I am assuming you're going to be adding more tests here. Things I'm curious to see:
* Tests using `storage.var = ...`
* Tests using `asm` blocks directly (when that's supported)
* Tests using complex control flow
* Tests having multiple contract calls or multiple storage writes in various scenarios. Do we emit multiple warnings in this case?

Reply:
We emit multiple warnings if there are multiple methods with CEI pattern violations, just like in test/src/e2e_vm_tests/test_programs/should_pass/static_analysis/cei_pattern_violation_english_auction/src/main.sw.

But for one method we report only the first violation, otherwise it can be very verbose, basically reporting each storage modifying statement after a contract call.